w3cdatalayer/application/factory.go

```diff
@@ -65,6 +67,13 @@ func (s *Factory) Inject(
 	s.locale = config.Locale
 	s.defaultCurrency = config.DefaultCurrency
 	s.hashUserValues = config.HashUserValues
+
+	regexString := "[,|;|\\|]"
+	r, err := regexp.Compile(regexString)
+	if err != nil {
+		panic(err)
```

Comment:
Do we really need a panic here? It'll ultimately result in crashing the whole app. Is this really the desired behavior?

Reply:
If that can answer your question, see the discussion with Thomas here https://gitlab.aoe.com/shared/i-love-flamingo/flamingo-commerce/merge_requests/58#note_44273

Reply:
@chetan-thapliyal I agree with you that `panic` should always be avoided. 
In this case we got a pretty good reason for the panic: If we have an erratic regex string, the regex won't compile and we won't get the expected functionality, but lots of possible error logs. 

So a better way would be to just panic at this point - as an erratic regex is equivalent to broken code.